src/server/procedure/procedure-with-graphql-client.ts

```diff
@@ -0,0 +1,22 @@
+import { MiddlewareFunction, TRPCError } from "@trpc/server";
+import { middleware, procedure } from "../server";
+import { saleorApp } from "../../../saleor-app";
+import { createClient } from "../../lib/graphql";
+import { attachAppToken } from "../middleware/attach-app-token";
+
+/**
+ * Construct common graphQL client and attach it to the context
+ */
+export const procedureWithGraphqlClient = procedure
```

Comment:
👌

Reply:
We could also add SaleorApp and SettingsManager.
Maybe the context should reflect the one we already have in the SDK?
https://github.com/saleor/saleor-app-sdk/blob/main/src/handlers/next/process-async-saleor-webhook.ts#L46
Probably it could make existing handlers/utils easier to use with tRPC?

Reply:
Hmm so probably its better to created a procedure with "everything" because its not that easy to compose middleware with tRPC. They work inline but you can't write middleware that assumes its after another middleware.

So we need something like "saleorAppProcedure" that will:
1. Have all context included like Next handlers
2. Be able to configure permissions (require token, require permission)

Since this is an example and I don't have real use case, I recommend that we keep it simple *now*, then rewrite one of our apps to tRPC and later extract this code to example. Maybe, having a real-life app (like search hub) with tRPC we can even remove synthetic example later and link to this app as an example

WDYT?

Reply:
> I recommend that we keep it simple now
Agree!

> like search hub
I see what you did there ;) 

>  rewrite one of our apps to tRPC and later extract this code to example
Also agree - till we use the library in real project, we wont be able to validate its strengths and issues. Would love to migrate search to tRPC since it has a nice integration with the react hook form.